src/ngircd/irc-channel.c

```diff
@@ -236,8 +237,56 @@ join_forward(CLIENT *Client, CLIENT *target, CHANNEL *chan,
 	}
 
 	/* tell users in this channel about the new client */
+		#define IMPLEMENT_EXTENDED_JOIN
```

Comment:
Please have preprocessor directives (`#define`) start in the first column: old(er) compilers don't support them being indented. This affects those lines here, but you have some down below, too.

**But**: Why do we need `IMPLEMENT_EXTENDED_JOIN` in the first place? I think you can just get rid of it altogether.